lib/uni/types.go

```diff
@@ -0,0 +1,68 @@
+// Package uni provides universal generic blockchain type abstractions.
+package uni
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+type Kind int
+
+const (
+	KindUnknown Kind = iota
+	KindEth          // eth/evm
+	KindSol          // solana
```

Comment:
Initial kinds are eth and sol. Should maybe rename to evm and svm?

Reply:
yeah i like EVM and SVM